src/debuginfo.cpp

```diff
@@ -811,14 +831,17 @@ void jl_getFunctionInfo(const char **name, size_t *line, const char **filename,
         if ((*it).second.func) {
             DISubprogram debugscope =
                 DISubprogram(prev.Loc.getScope((*it).second.func->getContext()));
-            *filename = debugscope.getFilename().data();
+            jl_copy_str(filename, debugscope.getFilename().data());
```

Comment:
I'm not sure if these `StringRef`s would actually leak: they don't own the underlying buffer and AFAIK `.data()` shouldn't need to allocate. (but valgrind might not be able to figure that out?)

Also, we should probably be more careful about copying from `StringRef` because they [are not guaranteed](http://llvm.org/docs/doxygen/html/classllvm_1_1StringRef.html#details) to be null-terminated.